docs/synapse/userguides/storm_ref_pivot.rstorm

```diff
@@ -7,1027 +7,1721 @@
+(``:zone`` and ``:ns``), Synapse will pivot to the FQDNs associated with both values. If you only want to
+pivot to the FQDNs associated with the name server (NS) FQDNs (for example), you must use explicit syntax:
 
+::
+  
+  <inet:dns:ns nodes> :ns -> inet:fqdn
 
-*Traverse the "hasip" light edge from an IPv4 address to the CIDR block(s) the IP is part of:*
 
+*Pivot from a set of X509 certificate metadata nodes (crypto:x509:cert nodes) to the associated SHA1 fingerprints (hash:sha1 nodes) and to any FQDNs (inet:fqdn nodes) associated with the certificates:*
 
-.. storm-pre:: inet:ipv4 <(hasip)- inet:cidr4
-::
+.. storm-pre:: [ crypto:x509:cert=* :md5=6b9fdcadf7ea5de1f1402a1cb62c7a65 :sha1=0008e9f15b9ff20de82c276cc54ec9dc094f54f8 :sha256=b1ee0920c4f4d5ded50732d15186eae5357eb743cbbb98ee935bd0791520b5f7 :identities:fqdns=(aliceplants.com,) ]
+.. storm-pre:: crypto:x509:cert -> ( hash:sha1, inet:fqdn )
 
-    <inet:ipv4> <(hasip)- inet:cidr4
+::
+  
+  <crypto:x509:cert nodes> -> ( hash:sha1, inet:fqdn )
 
+.. TIP::
+  
+  Specifying a list of target forms allows you to perform a more focused pivot (in contrast to pivoting to
+  any / all target forms using a wildcard). The elements of the list must be enclosed in parentheses and
+  separated by commas.
 
-*Traverse the "hasip" and "ipwhois" light edges from an IPv4 address to any nodes linked via those light edges (i.e., typically the CIDR block(s) the IP is part of and the netblock registration record(s) for the IP):*
+*Pivot from a set of X509 certificate metadata nodes (crypto:x509:cert nodes) to any/all nodes associated with any of the certificates' secondary properties:*
 
+.. storm-pre:: crypto:x509:cert -> *
 
-.. storm-pre:: inet:ipv4 <((hasip, ipwhois))- *
 ::
 
-    <inet:ipv4> <((hasip, ipwhois))- *
+  <crypto:x509:cert nodes> -> *
 
+The query above is an example of a **wildcard pivot out**. For any secondary properties on the source nodes,
+the query will return the associated nodes. For example, if the ``crypto:x509:cert:identities:ipv4s`` property
+is set, the query will return the associated ``inet:ipv4`` nodes. A wildcard pivot out is also known as a "refs
+out" pivot (for "references") because it pivots to the nodes "referenced by" the source nodes' secondary properties.
 
-*Traverse any / all light edges from an article to all nodes linked by any light edge:*
 
+.. _pivot-secondary-secondary:
 
-.. storm-pre:: media:news -(*)> *
-::
+Secondary to Secondary Property Pivot
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    <media:news> -(*)> *
+When pivoting from a secondary property of a set of source nodes to target nodes with the same secondary
+property (e.g., from the ``:ipv4`` property of a set of DNS A nodes to a set of network flow nodes with
+the same IPv4 as a ``:dst:ipv4`` property), the target can be:
 
+- a form name and property name; or
+- an interface name and property name.
```

Comment:
can also be a list of props like:
`inet:dns:a:ipv4=1.2.3.4 :ipv4 -> (inet:banner:server:ipv4, inet:flow:src:ipv4)`

Reply:
Updated